Allow setting DHCP_HOSTNAME on dynamic interfaces.

jmkeyes opened the pull request:
This patch allows setting the DHCP_HOSTNAME variable within an interface configuration file. When bringing up an interface with DHCP, the value of this variable will be used as the client's hostname when requesting a DHCP lease. It also targets the correct upstream branch. :+1:

razorsedge commented:
This looks great.  Might I ask for one last thing?  Can you update the README.md with some details of how to use this setting?

jmkeyes commented:
I've modified the README: I added the parameter to the expanded `network::if::dynamic` resource and added a note near the top mentioning that this module can now use it.

razorsedge commented:
Excellent! Thank you.
